### PR TITLE
Fix model type unit test

### PR DESCRIPTION
**Pull request by qiaozha**
fix #1526

### CODE REVIEW

File: packages/cadl-rlc-emitter/test/unit/generate/models.spec.ts [hunk 1]
```diff
@@ -310,42 +307,38 @@ describe("Input/output model type", () => {
...
         outputType
       });
     });
   });
-  // Error thrown
-  // Issue tracked here https://github.com/Azure/autorest.typescript/issues/1524
   describe("record generation", () => {
-    xit("should handle Record<int32> -> string", async () => {
-      await verifyPropertyType("Record<int32>", "string");
+    it("should handle Record<int32> -> Record<string, number>", async () => {
```
Comment (MaryGao) [inline packages/cadl-rlc-emitter/test/unit/generate/models.spec.ts:334]: Could you add a test case for Record<SimpleModel>?

**Review by MaryGao — APPROVED** [review]
Thanks for fixing these issues! Could add cases if missing?

File: packages/cadl-rlc-emitter/test/unit/generate/models.spec.ts [hunk 2]
```diff
@@ -310,44 +337,68 @@ describe("Input/output model type", () => {
...
+      await verifyPropertyType("Record<boolean>", "Record<string, boolean>");
     });
-    xit("should handle Record<string> -> string", async () => {
-      await verifyPropertyType("Record<string>", "string");
+    it("should handle Record<string> -> Record<string, string>", async () => {
+      await verifyPropertyType("Record<string>", "Record<string, string>");
     });
   });
+
+  describe("Record Model generation", () => {
```
Comment (qiaozha) [inline packages/cadl-rlc-emitter/test/unit/generate/models.spec.ts:375]: @MaryGao added here.

File: packages/cadl-rlc-emitter/test/unit/generate/models.spec.ts [hunk 3]
```diff
@@ -175,9 +174,39 @@ describe("Input/output model type", () => {
...
+      const cadlType = "plainDate[]";
+      const inputType = "Date[] | string[]";
+      const outputType = "string[]"
+      await verifyPropertyType(cadlType, inputType, { outputType });
+    });
   });
   describe("array models generation", () => {
-    // TODO: please add test cases here
+    it("should handle SimpleModel[] -> Array<SimpleModel>", async () => {
+      const cadlDefinition = `
```
Comment (MaryGao) [inline packages/cadl-rlc-emitter/test/unit/generate/models.spec.ts:187]: Do you know why the object array is `Array<SimpleModel>` not `SimpleModel[]`?

**Pull request merged**